modules/llrt_crypto/src/sha_hash.rs

```diff
@@ -195,8 +197,15 @@ pub struct ShaHash {
 #[rquickjs::methods]
 impl ShaHash {
     #[qjs(skip)]
-    pub fn new(algorithm: ShaAlgorithm, secret: Opt<ObjectBytes<'_>>) -> Result<Self> {
-        let secret = secret.0.map(|bytes| bytes.into_bytes());
+    pub fn new<'js>(
+        ctx: &Ctx<'js>,
+        algorithm: ShaAlgorithm,
+        secret: Opt<ObjectBytes<'js>>,
+    ) -> Result<Self> {
+        let secret = secret
+            .0
+            .map(|bytes| bytes.into_bytes(ctx))
+            .and_then(|res| res.ok());
```

Comment:
That is effectively silencing the error which is likely not what we want use transpose instead and return the error